### DISCUSSION TITLE
编译index.ts 报了一堆错，是这个包@wechatferry/agent有问题吗？

### DISCUSSION THREAD

**Discussion by dogscar** [wechatferry/wechatferry#6] [General]
index.ts:7:7 - error TS2339: Property 'on' does not exist on type 'WechatferryAgent'.

7 agent.on('message', (msg) => {
        ~~

node_modules/.pnpm/@types+node@16.9.1/node_modules/@types/node/globals.d.ts:72:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'AbortSignal' must be of type '{ new (): AbortSignal; prototype: AbortSignal; abort(reason?: any): AbortSignal; any(signals: AbortSignal[]): AbortSignal; timeout(milliseconds: number): AbortSignal; }', but here has type '{ new (): AbortSignal; prototype: AbortSignal; }'.

72 declare var AbortSignal: {
               ~~~~~~~~~~~

  node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/lib.dom.d.ts:2363:13
    2363 declare var AbortSignal: {
                     ~~~~~~~~~~~
    'AbortSignal' was also declared here.

node_modules/.pnpm/@types+node@16.9.1/node_modules/@types/node/module.d.ts:106:13 - error TS2386: Overload signatures must all be optional or required.

106             resolve?(specified: string, parent?: string | URL): Promise<string>;
                ~~~~~~~

node_modules/.pnpm/@wechatferry+agent@0.0.18/node_modules/@wechatferry/agent/dist/index.d.ts:3:8 - error TS1259: Module '"node:events"' can only be default-imported using the 'esModuleInterop' flag

3 import EventEmitter from 'node:events';
         ~~~~~~~~~~~~

  node_modules/.pnpm/@types+node@16.9.1/node_modules/@types/node/events.d.ts:622:5
    622     export = events;
            ~~~~~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1:8 - error TS1259: Module '"node:events"' can only be default-imported using the 'esModuleInterop' flag

1 import EventEmitter, { EventEmitter as EventEmitter$1 } from 'node:events';
         ~~~~~~~~~~~~

  node_modules/.pnpm/@types+node@16.9.1/node_modules/@types/node/events.d.ts:622:5
    622     export = events;
            ~~~~~~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:49:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

49         #private;
           ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:460:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

460         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:640:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

640         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:651:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

651         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:725:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

725         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:754:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

754         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:778:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

778         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:812:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

812         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:835:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

835         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:889:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

889         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:908:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

908         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:927:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

927         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:951:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

951         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:970:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

970         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:994:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

994         #private;
            ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1023:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1023         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1042:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1042         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1061:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1061         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1090:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1090         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1114:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1114         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1148:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1148         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1172:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1172         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1201:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1201         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1230:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1230         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1254:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1254         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1303:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1303         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1327:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1327         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1351:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1351         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1524:35 - error TS2315: Type 'EventEmitter' is not generic.

1524 declare class Wechatferry extends EventEmitter$1<WechatferryEventMap> {
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1782:5 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1782     #private;
         ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1832:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1832         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1869:5 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1869     #private;
         ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1903:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1903         #private;
             ~~~~~~~~

node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1927:9 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

1927         #private;
             ~~~~~~~~

node_modules/.pnpm/file-box@1.4.15/node_modules/file-box/dist/esm/src/file-box.type.d.ts:2:13 - error TS1192: Module '"http"' has no default export.

2 import type http from 'http';
              ~~~~


Found 40 errors in 6 files.

Errors  Files
     1  index.ts:7
     1  node_modules/.pnpm/@types+node@16.9.1/node_modules/@types/node/globals.d.ts:72
     1  node_modules/.pnpm/@types+node@16.9.1/node_modules/@types/node/module.d.ts:106
     1  node_modules/.pnpm/@wechatferry+agent@0.0.18/node_modules/@wechatferry/agent/dist/index.d.ts:3
    35  node_modules/.pnpm/@wechatferry+core@0.0.18/node_modules/@wechatferry/core/dist/index.d.ts:1
     1  node_modules/.pnpm/file-box@1.4.15/node_modules/file-box/dist/esm/src/file-box.type.d.ts:2

**Comment by KeJunMao** [comment]
纯 esm 模块，请使用 node18+，最好 node20

**Comment by dogscar** [comment]
node 版本 v20.17.0

**Reply by KeJunMao** [reply]
可以看下 example 文件夹
